src/SConscript

```diff
@@ -36,17 +36,15 @@ debug = ARGUMENTS.get('debug', 0)
 # Build flags in normal or debug mode
 if int(debug):
     env.Append(CPPFLAGS = ['-g', '-Wall', '-O0', '-finline-functions'])
+elif optflags:
+    env.Append( CPPFLAGS = list(optflags.split(" ")) )
 else:
     env.Append(CPPFLAGS = ['-O6', '-funroll-loops',
                            '-fomit-frame-pointer',
                            '-ffast-math', '-finline-functions',
                            '-fexpensive-optimizations'])
 
-# Default defines and optional flags
-if optflags:
-    env.Append(CPPFLAGS = optflags)
-else:
-    env.Append(CPPFLAGS = ['-DHAVE_LIBXPM', '-DHAVE_XF86_VIDMODE'])
+env.Append(CPPFLAGS = ['-DHAVE_LIBXPM', '-DHAVE_XF86_VIDMODE'])
```

Comment:
I see that setting optflags is very much broken for anything with more than 1 flag (missing `split`), but this makes it not possible to compile without these -DHAVE default flags, which would be always added after this patch.

--optflags is meant for additional CPPFLAGS mostly mean't to enable additional `#define`s, not to replace the defaults compiler optimizations though.

If that is meant to change (what is fedora trying to do?), then we should have a more consistent approach. If it is just about adding extra flags, then we should just add the list+split to the line where it is missing (and avoid altering the current behaviour).